### ISSUE TITLE
VST changes preset value on load

### ISSUE THREAD

**Issue by oeai** [LMMS/lmms#3847]
i'm not sure if this bugs vst, vestige or sound core
when i open in windows my vst that is bound to hardware i hear very different sound, it's not the same as i hear in another DAW with ASIO support (not WDM or else). 
i didn't mention if it asks me to save preset, if yes then it changes, i'll test it.
is there some kind of save latest config for vst? and if it tries to apply, then this can lead to a crash, because some values can be dynamically changed with using different synths + presets (+fx-fx)

the other way it can change, if it is reading midi and tries to send values somehow, while midi is not enabled for the channel

**Comment by oeai** [comment]
well, what I see is when I change presets inside of vst, values in the window of lmms managing-vst are not changing, so it is not reading from vst\preset, 
only if I press vst-sync button it changes.
so if it is changing output in anyway, then those changed inside of vst params will not actually change
solution is in auto-sync depending on choosed window, since all apps opens inside of lmms-desk

**Comment by zonkmachine** [comment]
Is this on the latest lmms-1.2.0-RC4 ?

**Comment by oeai** [comment]
it was rc3, but i've tested rc4 just now it works the same way. and automatisation button hides all knobs for furst time, then unhides, but nothing changes on third push

**Comment by musikBear** [comment]
@oeai you mention 'VST' so does this mean that ALL VSTs has this fail, or is this *one* specific VST
Please try one on the testetVST-list

Regarding the sync button: YES and yes! 👍  Bi-directional sync between both the VST UI and LMMS-dial-array would be marvellous, and it would fix all VST-issues emediately, because then lmms would only need to save *its own arrays values* No more nonsense with the dreathed **chunk**
The feature could be default in every save event, not controlled or influenced by user at all. -Eg, blackboxedI 
Already suggested earlier, btw

**Comment by oeai** [comment]
Tyrell works same way
Zebralette not reading presets to lmms at all
VST3 - no .dll, can't

**Comment by Umcaruje** [comment]
Those are both u-he vst's, does this happen with some vst's that are not
from them?

**Comment by oeai** [comment]
first synth was Proteus

**Comment by musikBear** [comment]
@oeai > device window

Wait one secd -Are you referring to the difference between selecting a VST-patch in the VST-UI, and selecting it in LMMS' UI?
It is true, LMMS will not recognize a change of patch *graphically* if it is made in the VST-UI, but it is changed *logically*, and for the VSTs in the *tested-list* the patch is saved inside the project, but the name is shown as 1. patch-option on the LMMS UI, however correct patch-name is shown inside the VST
@Umcaruje That behaviour is also true for *Synth1*, and many more

**Comment by oeai** [comment]
@musikBear yes, i'm about that, but I don't know which UI will override another (if I save or render) because of that.

**Comment by musikBear** [comment]
@oeai imo this is only graphical, not logical. What i think is that the VSTs you have problems with, simply does not pass the 5 point test:
https://lmms.io/wiki/index.php?title=Working_with_VSTs#The_five_point_VST_test
The behaviour you describe, is also in Synth1, but the chosen patch is preserved and reloaded with all user settings.
The fact that the name is not updated in LMMS ui is correct.

**Comment by DomClark** [comment]
It's not entirely clear to me quite what this issue is about, but it sounds like it might have been fixed by #4401.
@oeai Can you test rc7 to see if this bug still persists?

**Comment by musikBear** [comment]
@DomClark  I did it for Tyrell
a change of preset in VSTUI does not showup as a change in LMMS-UI, however is **is** changed!
Not updating the preset-name is at most a glitch, imo
An interesting thing is, that after project-reload, there is no options for selecting presets from LMMS-UI. The Select only has one item!
If a new preset is VST-selected, project saved and reloaded, it is the *new* preset, that is *correctly* listed as preset-name on Preset-display in LMMS-UI.
Bottom-line: No synch from preset-select in VST-UI with LMMS-UI
Changes on VSTUI**are** preserved (goes for checkbox, selects, sliders and dials) -Eg pass 5 point-test and are fully functional.

**Comment by DomClark** [comment]
TyrellN6 doesn't seem to present its presets to the host in a normal manner; I've tried this in other hosts and the only available preset is the one that is currently selected. The preset name not updating is because LMMS doesn't think the preset has changed; since the plugin only exposes a single preset at once, it is always the first preset that is selected. I could fix this, but the current system for sending VST information back to LMMS is rather restricted, and since it's only a minor cosmetic issue it's not high priority for 1.2. Hopefully I (or somebody else) will get round to this for 1.3.

**Comment by Spekular** [comment]
Closing and consolidating into #5433
